File: Medium/0775-Global and Local Inversions/Solution.rb
# @param {Integer[]} nums
# @return {Boolean}
def is_ideal_permutation(nums)
  max = 0

  (2...nums.size).each do |i|
    max = [max, nums[i - 2]].max
    return false if nums[i] < max
  end

  true
end

File: Medium/0775-Global and Local Inversions/Solution.rs
impl Solution {
    pub fn is_ideal_permutation(nums: Vec<i32>) -> bool {
        let mut max = 0;

        for i in 2..nums.len() {
            max = max.max(nums[i - 2]);
            if nums[i] < max {
                return false;
            }
        }

        true
    }
}
